src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,25 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  books: [],
+};
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.books.push(action.payload);
+    },
```

Comment:
- [ ] **[OPTIONAL]** The `push()` method also mutates the array, however we can avoid any side effects by using the spread operator.

  ```
  addBook: (state, action) => {
    state.books = [
      ...state.books,
      {
        id: Math.floor(Math.random() * 100),
        title: action.payload.title,
        author: action.payload.author,
      },
    ]
  },
  ```